org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/codemanipulation/StubUtility.java

```diff
@@ -204,6 +207,47 @@ public static String getCatchBodyContent(ICompilationUnit cu, String exceptionTy
 		context.setVariable(CodeTemplateContextType.EXCEPTION_VAR, variableName);
 		return evaluateTemplate(context, template);
 	}
+
+	public static String getClassSnippetContent(ICompilationUnit cu, String lineDelimiter) throws CoreException {
+		CodeGenerationTemplate templateSetting = CodeGenerationTemplate.CLASSSNIPPET;
+		Template template = templateSetting.createTemplate(cu.getJavaProject());
+		if (template == null) {
+			return null;
+		}
+		CodeTemplateContext context = new CodeTemplateContext(template.getContextTypeId(), cu.getJavaProject(), lineDelimiter);
+		setVariables(context, cu);
+		return evaluateTemplate(context, template);
+	}
+
+	public static String getInterfaceSnippetContent(ICompilationUnit cu, String lineDelimiter) throws CoreException {
+		CodeGenerationTemplate templateSetting = CodeGenerationTemplate.INTERFACESNIPPET;
+		Template template = templateSetting.createTemplate(cu.getJavaProject());
+		if (template == null) {
+			return null;
+		}
+		CodeTemplateContext context = new CodeTemplateContext(template.getContextTypeId(), cu.getJavaProject(), lineDelimiter);
+		setVariables(context, cu);
+		return evaluateTemplate(context, template);
+	}
+
+	private static void setVariables(CodeTemplateContext context, ICompilationUnit cu) throws CoreException {
+		IPackageDeclaration[] packageDeclartions = cu.getPackageDeclarations();
+		String packageHeader = (packageDeclartions == null || packageDeclartions.length == 0) ? "package " + cu.getParent().getElementName() + ";\n\n" : "";
+		context.setVariable(CodeTemplateContextType.PACKAGEHEADER, packageHeader);
+		String typeName = JavaCore.removeJavaLikeExtension(cu.getElementName());
+		List<IType> types = Arrays.asList(cu.getTypes());
+		int postfix = 0;
+		while (types != null && !types.isEmpty() && types.stream().filter(isTypeExists(typeName)).findFirst().isPresent()) {
+			typeName = "Inner" + JavaCore.removeJavaLikeExtension(cu.getElementName()) + (postfix == 0 ? "" : "_" + postfix);
+			postfix++;
+		}
```

Comment:
minor annoyance but inner inner class doesn't work:
<img width="359" alt="screen shot 2018-05-28 at 3 29 32 pm" src="https://user-images.githubusercontent.com/148698/40627292-26f3798e-628c-11e8-90c9-06c829d414b1.png">
We're not getting the proper context here. completion should know it's inside the inner class already